### PR TITLE
Basic Set up for the plugin

### PR DESCRIPTION
**Pull request by Sak1012**
This PR contains the basic setup of the exhibitor plugin

## Summary by Sourcery

Set up the basic structure for the Exhibitor plugin, including CI workflows for style checks and testing, a script for managing translations, and initial configuration files for development and deployment.

New Features:
- Introduce the Exhibitor plugin for Pretix, allowing users to add and control exhibitors in events.

CI:
- Add GitHub Actions workflows for code style checks using isort, flake8, and black.
- Add GitHub Actions workflow for running tests with pytest and coverage.
- Add GitLab CI configuration for testing, style checks, and PyPI deployment.

Deployment:
- Set up a script for updating locale files and managing translations with Weblate.

Tests:
- Add a basic test file with a placeholder test to verify the testing setup.

Chores:
- Add a Makefile for compiling and generating locale messages.
- Add a setup configuration file for flake8, isort, pytest, and coverage settings.
- Add a script to install Git hooks for pre-commit checks using black, isort, and flake8.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
+exhibitors
+==========================
+
+This is a plugin for `pretix`_.
```
Comment (mariobehling) [inline README.rst:4]: Change everything to eventyay-tickets

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    black .
+
+To automatically check for these issues before you commit, you can run ``.install-hooks``.
+
+
+License
+-------
+
+
+Copyright 2024 Srivatsav Auswin
```
Comment (mariobehling) [inline README.rst:46]: Change to FOSSASIA

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+Released under the terms of the Apache License 2.0
+
+
+
+.. _pretix: https://github.com/pretix/pretix
```
Comment (mariobehling) [inline README.rst:52]: ?

File: exhibitors/apps.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    raise RuntimeError("Please use pretix 2.7 or above to run this plugin!")
+
+
+class ExhibitorApp(PluginConfig):
+    name = gettext_lazy("exhibitors")
+    verbose_name = gettext_lazy("exhibitors")
+
+    class PretixPluginMeta:
+        name = gettext_lazy("exhibitors")
+        author = "Srivatsav Auswin"
```
Comment (mariobehling) [inline exhibitors/apps.py:17]: FOSSASIA

File: exhibitors/apps.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        description = gettext_lazy("This plugin enables to add and control exhibitors in eventyay")
+        visible = True
+        version = __version__
+        category = "FEATURE"
+        compatibility = "pretix>=2.7.0"
```
Comment (mariobehling) [inline exhibitors/apps.py:22]: eventyay-tickets

File: exhibitors/apps.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
+from django.utils.translation import gettext_lazy
+
+from . import __version__
+
+try:
+    from pretix.base.plugins import PluginConfig
+except ImportError:
+    raise RuntimeError("Please use pretix 2.7 or above to run this plugin!")
```
Comment (mariobehling) [inline exhibitors/apps.py:8]: eventyay-tickets. "Please use a later version of eventyay-tickets."

File: README.rst [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+.. _pretix development setup: https://docs.pretix.eu/en/latest/development/setup.html
```
Comment (mariobehling) [inline README.rst:53]: ?

**Review by mariobehling — CHANGES_REQUESTED** [review]
Definitely look and fix all files you are submitting here and ensure the project does not have other references except eventyay-tickets. A simple look at "Files Changed" will already help you to get started.

File: .github/workflows/style.yml [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.11
+      - uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/setup.py') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - name: Install pretix
```
Comment (mariobehling) [inline .github/workflows/style.yml:52]: can this be changed or not?

File: .github/workflows/style.yml [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+        run: pip3 install pretix
```
Comment (mariobehling) [inline .github/workflows/style.yml:53]: ?

File: .github/workflows/style.yml [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.11
+      - uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/setup.py') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - name: Install pretix
```
Comment (mariobehling) [inline .github/workflows/style.yml:74]: ?

File: .gitlab-ci.yml [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+        - uv pip install --system -U wheel setuptools pytest pytest-django coverage
+        - uv pip install --system -U git+https://github.com/pretix/pretix.git@master#egg=pretix
+    script:
+        - uv pip install --system -e .
+        - make
+        - coverage run -m pytest tests
+        - coverage report
+style:
+    image:
+        name: pretix/ci-image
```
Comment (mariobehling) [inline .gitlab-ci.yml:15]: ?

File: setup.cfg [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+known_standard_library = typing
+multi_line_output = 5
+not_skip = __init__.py
+skip = setup.py
+
+[tool:pytest]
+DJANGO_SETTINGS_MODULE = pretix.testutils.settings
+
+[coverage:run]
+source = pretix_adyen
```
Comment (mariobehling) [inline setup.cfg:20]: ?

File: setup.cfg [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+default_section = THIRDPARTY
+include_trailing_comma = true
+known_third_party = pretix
+known_standard_library = typing
+multi_line_output = 5
+not_skip = __init__.py
+skip = setup.py
+
+[tool:pytest]
+DJANGO_SETTINGS_MODULE = pretix.testutils.settings
```
Comment (mariobehling) [inline setup.cfg:17]: ?

File: setup.cfg [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+    NOQA
+    NotImplementedError
+
+[check-manifest]
+ignore =
+    .update-locales
+    .update-locales.sh
+    .gitlab-ci.yml
+    .install-hooks.sh
+    pretixplugin.toml
```
Comment (mariobehling) [inline setup.cfg:38]: ?

**Comment by Sak1012** [comment]
Added basic signals to enable plugin and see it in the nav bar with a separate settings section and a edit section

**edit section**
image

**settings section**
image

File: exhibitors/urls.py [hunk 15]
```diff
@@ -0,0 +1,18 @@
+from django.urls import re_path as urls
+
+from .views import SettingsView, ExhibitorListView, ExhibitorCreateView, ExhibitorEditView, ExhibitorDeleteView, ExhibitorCopyKeyView
+
+urlpatterns = [
+    urls(r'^control/event/(?P<organizer>[^/]+)/(?P<event>[^/]+)/settings/exhibitors',
```
Comment (hongquan) [inline exhibitors/urls.py:6]: Please use new style of URL conf https://docs.djangoproject.com/en/5.1/ref/urls/#path

File: pyproject.toml [hunk 16]
```diff
@@ -0,0 +1,43 @@
+[project]
+name = "exhibitors"
+dynamic = ["version"]
+description = "Eventyay-tickets plugin to add and manage exhibitors"
+readme = "README.rst"
+requires-python = ">=3.9"
```
Comment (hongquan) [inline pyproject.toml:6]: Do we need to support Python as old as v3.9?

**Review by hongquan — CHANGES_REQUESTED** [review]

File: exhibitors/models.py [hunk 17]
```diff
@@ -0,0 +1,56 @@
+from django.db import models
+import string
+import json
+import random
+import os
+from django.utils.translation import gettext_lazy as _
+from pretix.base.models import Event
+
+def generate_key():
+    return ''.join(random.choices(
```
[the remark this answers is not in the document]
Comment (Sak1012) [reply exhibitors/models.py:10]: Yes, this is now implemented for testing purpose will be changed later

File: exhibitors/urls.py (hunk 15 above)
Comment (Sak1012) [reply exhibitors/urls.py:6]: This was the format of urls.py for other plugins in eventyay-tickets hence followed the same. Should I change to the new style ?

File: pyproject.toml (hunk 16 above)
Comment (Sak1012) [reply pyproject.toml:6]: I just added it as a basic requirement can be changed to a later version

**Comment by Sak1012** [comment]
when a GET request is sent to `/api/v1/organizers/<organiser>/events/<slug>/exhibitors/` with Accept and Token headers a response of the exhibitor details can be seen
```json
{
    "count": 2,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 1,
            "name": "Google",
            "description": "something there",
            "url": "https://www.google.com/",
            "email": "google@gmail.com",
            "logo": "http://localhost:8000/media/exhibitors/logos/Google/mock.png",
            "key": "ch4de60x",
            "lead_scanning_enabled": true
        },
        {
            "id": 3,
            "name": "Microsoft",
            "description": "details here",
            "url": "https://photos.google.com/",
            "email": "auswinsrivatsav@gmail.com",
            "logo": "http://localhost:8000/media/exhibitors/logos/Microsoft/mock.png",
            "key": "2pda7hlf",
            "lead_scanning_enabled": true
        }
    ]
}
```

**Note:** The `key` section will be later removed, it's just there for testing.

**Comment by Sak1012** [comment]
Implemented a basic auth api that can be used by Open-Event-Checkin for Exhibitor login
image
**Note:** This is a base API more proper authentication functions will be added later

File: pyproject.toml (hunk 16 above)
Comment (mariobehling) [reply pyproject.toml:6]: No need for 3.9. Let's stick to newer versions.

Comment (Sak1012) [reply pyproject.toml]: I have updated it to **3.11**

File: .update-locales.sh [hunk 18]
```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
```
[the remark this answers is not in the document]
Comment (hongquan) [reply .update-locales.sh:1]: Please follow this sourcery-ai suggestion.

File: .gitlab-ci.yml [hunk 19]
```diff
@@ -0,0 +1,42 @@
+test:
+    image:
+        name: pretix/ci-image
+    before_script:
+        - pip install -U pip uv
+        - uv pip install --system -U wheel setuptools pytest pytest-django coverage
+        - uv pip install --system -U git+https://github.com/pretix/pretix.git@master#egg=pretix
```
[the remark this answers is not in the document]
Comment (mariobehling) [reply .gitlab-ci.yml:7]: We definitely should not depend on the pretix repo. eventyay-tickets should be the dependency.

**Review by mariobehling — CHANGES_REQUESTED** [review]

**Review by mariobehling — CHANGES_REQUESTED** [review]
Please use Github Actions instead of Gitlab.ci. Lets stay on one platform.

File: .gitlab-ci.yml (hunk 19 above)
[the remark this answers is not in the document]
Comment (Sak1012) [reply .gitlab-ci.yml]: I have now changes it with a different setup based of eventyay-tickets

**Comment by mariobehling** [comment]
Thanks! To be done: Substitute Gitlab.ci with Github Actions.

File: .gitlab-ci.yml [hunk 20]
```diff
@@ -1,42 +1,65 @@
...
+    name: eventyay-tickets/ci-image
+  stage: release
+  before_script:
+    - cp /keys/.pypirc ~/.pypirc
+  script:
+    - pip install -U check-manifest twine
+    - XDG_CACHE_HOME=/cache pip install -e ".[dev]"
+    - cd src
+    - python setup.py sdist
+    - pip install dist/pretix-*.tar.gz
```
Comment (mariobehling) [inline .gitlab-ci.yml:51]: why?

**Review by mariobehling — CHANGES_REQUESTED** [review]

**Comment by Sak1012** [comment]
> Thanks! To be done: Substitute Gitlab.ci with Github Actions.

I have added all the functionality of gitlab-ci in workflow files.

File: pretixplugin.toml
Comment (mariobehling) [inline pretixplugin.toml:1]: why this cannot be part of the pyproject.toml?

**Review by mariobehling — CHANGES_REQUESTED** [review]

File: setup.cfg [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+    der __repr__
+    if settings.DEBUG
+    NOQA
+    NotImplementedError
+
+[check-manifest]
+ignore =
+    .update-locales
+    .update-locales.sh
+    .gitlab-ci.yml
```
Comment (mariobehling) [inline setup.cfg:36]: delete it please and check other parts of setup.cfg

**Review by mariobehling — CHANGES_REQUESTED** [review]

**Review by odkhang — CHANGES_REQUESTED** [review]

File: setup.cfg (hunk 21 above)
Comment (Sak1012) [reply setup.cfg]: Removed 👍

File: exhibitors/migrations/0001_initial.py [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+import django.db.models.deletion
+import exhibitors.models
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+        ('pretixbase', '0005_alter_cachedcombinedticket_id_alter_cachedticket_id_and_more'),
```
[the remark this answers is not in the document]
Comment (Sak1012) [reply exhibitors/migrations/0001_initial.py:13]: Reverted back to the previous migration file(004) that's available

**Review by odkhang — APPROVED** [review]

**Pull request merged**